core/exec/src/mill/exec/Execution.scala

```diff
@@ -14,18 +14,34 @@ import scala.concurrent._
 /**
  * Core logic of evaluating tasks, without any user-facing helper methods
  */
-private[mill] trait ExecutionCore extends GroupExecution {
-
-  def baseLogger: ColorLogger
-  protected[exec] def chromeProfileLogger: ChromeProfileLogger
-  protected[exec] def profileLogger: ProfileLogger
+private[mill] case class Execution(
```

Comment:
Shouldn't this be an normal non-case class, since some parameters refer to mutable data structures?